lib/streamlit/runtime/metrics_util.py

```diff
@@ -72,3 +112,193 @@ def __repr__(self) -> str:
     @property
     def installation_id(self):
         return self.installation_id_v3
+
+
+def _get_type_name(obj: object) -> str:
+    with contextlib.suppress(Exception):
+        obj_type = type(obj)
+        type_name = "unknown"
+        if hasattr(obj_type, "__qualname__"):
+            type_name = obj_type.__qualname__
+        elif hasattr(obj_type, "__name__"):
+            type_name = obj_type.__name__
+
+        if obj_type.__module__ != "builtins":
+            # Add the full module path
+            type_name = f"{obj_type.__module__}.{type_name}"
+
+        if type_name in _OBJECT_NAME_MAPPING:
+            type_name = _OBJECT_NAME_MAPPING[type_name]
+        return type_name
+    return "failed"
+
+
+def _get_callable_name(callable: Callable[..., Any]) -> str:
+    with contextlib.suppress(Exception):
+        name = "unknown"
+        if inspect.isclass(callable):
+            name = callable.__class__.__name__
+        elif hasattr(callable, "__qualname__"):
+            name = callable.__qualname__
+        elif hasattr(callable, "__name__"):
+            name = callable.__name__
+        if name in _CALLABLE_NAME_MAPPING:
+            name = _CALLABLE_NAME_MAPPING[name]
+        elif "." in name:
```

Comment:
Will this code ever be executed? If you have `name = _CALLABLE_NAME_MAPPING[name]`, it looks like none of the callable names have periods in them.

Reply:
Yep, most of the other commands actually have a path prefix based on the mixing class.